### PR TITLE
import Protocol No

### PR DESCRIPTION
**Pull request by victoria34**
screen shot 2019-02-06 at 5 38 38 pm

### CODE REVIEW

File: src/main/webapp/app/trial/trial.component.ts [hunk 1]
```diff
@@ -250,6 +271,33 @@ export class TrialComponent implements OnInit, AfterViewInit {
     cancelUpdateNote() {
         this.noteEditable = false;
     }
+    updateProtocolNo() {
+        if ( this.trialChosen['protocol_no'].match( /^\d+-\d+$/g ) ) {
```
Comment (zhx828) [inline src/main/webapp/app/trial/trial.component.ts]: The Regex is a bit different comparing to the one above. Could you create some const to share?

File: src/main/webapp/app/trial/trial.component.ts [hunk 2]
```diff
@@ -250,6 +271,33 @@ export class TrialComponent implements OnInit, AfterViewInit {
...
+            const result = confirm('Are you sure to update Protocol No to ' + this.trialChosen['protocol_no'] + '?');
+            if (result) {
+                this.trialService.getRef( 'Trials/' + this.nctIdChosen ).update( {protocol_no: this.trialChosen['protocol_no']} )
+                .then((res) => {
+                    this.protocolNoMessage.content = 'Update Protocol No successfully.';
```
Comment (zhx828) [inline src/main/webapp/app/trial/trial.component.ts]: `No.` or `Update protocol number successfully.`

File: src/main/webapp/app/trial/trial.component.ts [hunk 3]
```diff
@@ -250,6 +271,33 @@ export class TrialComponent implements OnInit, AfterViewInit {
...
+                    this.protocolNoMessage.color = 'green';
+                })
+                .catch( ( error ) => {
+                    this.protocolNoMessage.content = 'Failed to update Protocol No.';
+                    this.protocolNoMessage.color = 'red';
+                    this.trialChosen['protocol_no'] = this.originalTrial['protocol_no'];
+                } );
+            }
+        } else {
+            this.protocolNoMessage.content = 'Protocol No should follow the format: number-number.';
```
Comment (zhx828) [inline src/main/webapp/app/trial/trial.component.ts]: Same here

File: src/main/webapp/app/trial/trial.component.ts (hunk 2 above)
Comment (victoria34) [reply src/main/webapp/app/trial/trial.component.ts]: Good point.

File: src/main/webapp/app/trial/trial.component.ts (hunk 1 above)
Comment (victoria34) [reply src/main/webapp/app/trial/trial.component.ts]: Sorry, I should use this one. I forgot to change the one above.

**Pull request merged**